pkg/polaris/ctl_polaris.go

```diff
@@ -67,317 +66,264 @@ type CRSpecBuilderFromCobraFlags struct {
+		case "postgres-username":
+			util.SetHelmValueInMap(ctl.args, []string{"postgres", "user"}, ctl.flagTree.PostgresUsername)
+		case "postgres-password":
+			if len(ctl.flagTree.PostgresPassword) == 0 {
+				log.Errorf("you must set postgres-password")
+				isErrorExist = true
 			}
-			ctl.spec.Licenses.Polaris = data
+			util.SetHelmValueInMap(ctl.args, []string{"postgres", "password"}, ctl.flagTree.PostgresPassword)
+		case "postgres-size":
+			util.SetHelmValueInMap(ctl.args, []string{"postgres", "size"}, ctl.flagTree.PostgresSize)
+		case "postgres-ssl-mode":
+			var sslMode PostgresSSLMode
+			switch PostgresSSLMode(ctl.flagTree.PostgresSSLMode) {
+			case PostgresSSLModeDisable:
+				sslMode = PostgresSSLModeDisable
+			//case PostgresSSLModeAllow:
+			//  ctl.args["postgres.sslMode"] = fmt.Sprintf("%s", PostgresSSLModeAllow)
+			//case PostgresSSLModePrefer:
+			//  ctl.args["postgres.sslMode"] = fmt.Sprintf("%s", PostgresSSLModePrefer)
+			case PostgresSSLModeRequire:
+				sslMode = PostgresSSLModeRequire
+			default:
+				log.Errorf("%s is an invalid value for --postgres-ssl-mode", ctl.flagTree.PostgresSSLMode)
+				isErrorExist = true
+			}
+			util.SetHelmValueInMap(ctl.args, []string{"postgres", "sslMode"}, sslMode)
 		default:
 			log.Debugf("flag '%s': NOT FOUND", f.Name)
 		}
 	} else {
 		log.Debugf("flag '%s': UNCHANGED", f.Name)
 	}
-}
-
-// GetPolarisDefault returns PolarisDB default configuration
-func GetPolarisDefault() *Polaris {
-	return &Polaris{
-		ImagePullSecrets:    "gcr-json-key",
-		IngressClass:        "nginx",
-		Licenses:            &Licenses{},
-		OrganizationDetails: &OrganizationDetails{},
-		EnableReporting:     false,
-		PolarisSpec: &PolarisSpec{
-			DownloadServerDetails: DownloadServerDetails{
-				Storage: Storage{
-					StorageSize: DOWNLOAD_SERVER_PV_SIZE,
-				},
-			},
-		},
-		ReportingSpec: &ReportingSpec{
-			ReportStorageDetails: ReportStorageDetails{
-				Storage: Storage{
-					StorageSize: REPORT_STORAGE_PV_SIZE,
-				},
-			},
-		},
-		PolarisDBSpec: &PolarisDBSpec{
-			SMTPDetails: SMTPDetails{
-				TLSCheckServerIdentity: true,
-				TLSMode:                SMTPTLSModeDisable,
-			},
-			PostgresDetails: PostgresDetails{
-				Host:       "postgresql",
-				Username:   "postgres",
-				Port:       5432,
-				IsInternal: false,
-				SSLMode:    PostgresSSLModeRequire,
-				Storage: Storage{
-					StorageSize: POSTGRES_PV_SIZE,
-				},
-			},
-			MongoDBDetails: MongoDBDetails{
-				Storage: Storage{
-					StorageSize: MONGODB_PV_SIZE,
-				},
-			},
-			EventstoreDetails: EventstoreDetails{
-				Replicas: util.IntToInt32(3),
-				Storage: Storage{
-					StorageSize: EVENTSTORE_PV_SIZE,
-				},
-			},
-			UploadServerDetails: UploadServerDetails{
-				Replicas: util.IntToInt32(1),
-				Storage: Storage{
-					StorageSize: UPLOAD_SERVER_PV_SIZE,
-				},
-			},
-		},
+	if isErrorExist {
```

Comment:
This will only catch one error since "isErrorExist" isn't a global variable. If you want it to error after checking all the flags then I think this check needs to be moved into the GenerateHelmFlagsFromCobraFlags() command

Reply:
Nice catch.. Will fix @mphammer